### PR TITLE
Convert module to PDK

### PR DESCRIPTION
**Pull request by nmaludy**
PDK is the Puppet Development Kit https://puppet.com/docs/pdk/1.x/pdk.html

This is the new standard for module structure.

This PR converts the `stackstorm/st2` module to be PDK compliant.

**DO NOT MERGE UNTIL https://github.com/StackStorm/puppet-st2/pull/235 IS MERGED**

### CODE REVIEW

**Comment by nmaludy** [comment]
@armab this is ready to go

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    PUPPET_GEM_VERSION: '~> 5.5'
+
+parallel_spec-Ruby 2.4.4-Puppet ~> 5.5:
+  stage: unit
+  image: ruby:2.4.4
+  script:
+    - bundle exec rake parallel_spec
+  variables:
+    PUPPET_GEM_VERSION: '~> 5.5'
+
```
Comment (armab) [inline .gitlab-ci.yml]: `gitlab-ci.yml`?
Copy-pasted from other place?

File: appveyor.yml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+  - gem -v
+  - bundle -v
+  - bundle exec rake %CHECK%
+notifications:
+  - provider: Email
+    to:
+      - nobody@nowhere.com
+    on_build_success: false
+    on_build_failure: false
+    on_build_status_changed: false
```
Comment (armab) [inline appveyor.yml]: Same for `appveyor.yml`. We don't run it, probably copy-pasted

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,118 @@
+---
+require: rubocop-rspec
```
Comment (armab) [inline .rubocop.yml]: We didn't enforce Rubocop linting before?

Makes sense to include that as a new improvement in CHANGELOG.

File: test/integration/stackstorm/controls/mongodb_test.rb [hunk 4]
```diff
@@ -24,7 +25,7 @@
     # Mongo should listen on localhost only
     # @link: https://github.com/StackStorm/st2-packages/blob/a93701d98a130f50f7cb551e842889212ece3b11/scripts/st2bootstrap-deb.sh#L446-L447
     its(['net.bindIp']) { should eq '127.0.0.1' }
-    its(['net.port']) { should eq 27017 }
+    its(['net.port']) { should eq 27_017 }
```
Comment (armab) [inline test/integration/stackstorm/controls/mongodb_test.rb]: Any details behind this particular change?

**Comment by nmaludy** [comment]
PDK is a tool that manages common files in a Puppet repository.
It allows developers to synchronize from a "template" repository so common configs like:
.gitignore, .travis.yml, rubocop, etc. Are all copied over and don't have to be written
from scratch. Most of the changes are new files, added by the `pdk update` command.

Any deviations in these files are tracked using the `.sync.yml` file. This way
all deviations are kept in source control and can be managed. For right now the
only major deviation is that we want to maintain our own `.travis.yml` for
the kitchen integration work.

Part of the templating that is done is Rakefiles used for unit testing.
Some of the small changes done in .travis.yml, Dockerfiles, and the build/scripts/ci*.sh
involve utilizing these new Rakefile tasks so we can be "compliant" with the PDK
testing standards.

File: build/centos6-puppet4/Gemfile.lock [hunk 5]
```diff
@@ -1,133 +0,0 @@
-GEM
-  remote: https://rubygems.org/
-  specs:
```
Comment (armab) [inline build/centos6-puppet4/Gemfile.lock]: Any background behind removing `Gemfile.lock` files?
How that can affect CI builds/reproducibility?

Comment (nmaludy) [reply build/centos6-puppet4/Gemfile.lock]: Both the Gemfile and Gemfile.lock's are no longer specific to each distribution (thanks to a standardized Ruby version in Puppet 4/5)

Instead the Gemfile from the repo root is used to defined gems for unit testing.

File: .rubocop.yml (hunk 3 above)
Comment (nmaludy) [reply .rubocop.yml]: Done!

File: test/integration/stackstorm/controls/mongodb_test.rb (hunk 4 above)
Comment (nmaludy) [reply test/integration/stackstorm/controls/mongodb_test.rb]: would you like me to try and disable this check?

Comment (armab) [reply test/integration/stackstorm/controls/mongodb_test.rb]: No, it's OK. Was just curious about this hack

**Pull request merged**